pkg/monitoring/metrics/forklift-controller/plan_metrics.go

```diff
@@ -0,0 +1,133 @@
+package forklift_controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Calculate Plans metrics every 10 seconds
+func RecordPlanMetrics(c client.Client) {
+	go func() {
+		for {
+			time.Sleep(10 * time.Second)
+
+			// get all migration objects
+			plans := api.PlanList{}
+			err := c.List(context.TODO(), &plans)
+
+			// if error occurs, retry 10 seconds later
+			if err != nil {
+				fmt.Printf("Metrics Plans list error: %v\n", err)
+				continue
+			}
+
+			// Holding counter vars used to make gauge update "atomic"
+			var (
+				succeededRHV, succeededOCP, succeededOVA, succeededVsphere, succeededOpenstack float64
+				failedRHV, failedOCP, failedOVA, failedVsphere, failedOpenstack                float64
+				localCluster, remoteCluster                                                    float64
+				warmMigration, coldMigration                                                   float64
+			)
+
+			for _, m := range plans.Items {
+
+				destProvider := api.Provider{}
+				err := c.Get(context.TODO(), client.ObjectKey{Namespace: m.Spec.Provider.Destination.Namespace, Name: m.Spec.Provider.Destination.Name}, &destProvider)
+				if err != nil {
+					continue
+				}
+				fmt.Println("this is provider ", destProvider)
+				if destProvider.Spec.URL == "" {
+					localCluster++
+				} else {
+					remoteCluster++
+				}
+
+				if m.Spec.Warm {
+					warmMigration++
+				} else {
+					coldMigration++
+				}
+
+				sourceProvider := api.Provider{}
+				err = c.Get(context.TODO(), client.ObjectKey{Namespace: m.Spec.Provider.Source.Namespace, Name: m.Spec.Provider.Source.Name}, &sourceProvider)
+				if err != nil {
+					continue
+				}
+				if m.Status.HasCondition(Succeeded) {
```

Comment:
what about running and pending(existing plan that didn't start to run)/cancel conditions?

Reply:
we can include it maybe for migrations only, we have a lot of statuses the questions which info is interesting for us.